core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapYearInComparison.java

```diff
@@ -146,67 +147,66 @@ public Expression rewriteInPredicate(InPredicate node, Void context, ExpressionT
             // unwrap the InPredicate only in case we manage to unwrap the entire value list
             ImmutableList.Builder<Expression> comparisonExpressions = ImmutableList.builderWithExpectedSize(inListExpression.getValues().size());
             for (Expression rightExpression : inListExpression.getValues()) {
-                ComparisonExpression comparisonExpression = new ComparisonExpression(EQUAL, value, rightExpression);
-                Expression unwrappedExpression = unwrapYear(comparisonExpression);
-                if (unwrappedExpression == comparisonExpression) {
+                Optional<Expression> unwrappedExpression = unwrapYear(EQUAL, value, rightExpression);
+                if (unwrappedExpression.isEmpty()) {
                     return inPredicate;
                 }
-                comparisonExpressions.add(unwrappedExpression);
+                comparisonExpressions.add(unwrappedExpression.get());
             }
 
             return or(comparisonExpressions.build());
         }
 
         // Simplify `year(d) ? value`
-        private Expression unwrapYear(ComparisonExpression expression)
+        private Optional<Expression> unwrapYear(ComparisonExpression.Operator operator, Expression leftExpression, Expression rightExpression)
```

Comment:
this is in `Review comments - to be squashed before merge` commit, but there is nowhere to squash this into.

it looks like @martint has comments that would best apply in https://github.com/trinodb/trino/pull/18138, so maybe let's extract `UnwrapYearInComparison` to separate PR (or at least a prep commit)

Reply:
it will be in a different commit of course. 
not sure a separate PR is needed.